savant/deepstream/meta/frame.py

```diff
@@ -40,6 +42,22 @@ def objects(self) -> Iterator[ObjectMeta]:
         """
         return NvDsObjectMetaIterator(self.frame_meta)
 
+    @property
+    def roi(self) -> BBox:
+        if not self._primary_obj:
+            for obj_meta in self.objects:
+                if obj_meta.is_primary:
+                    self._primary_obj = obj_meta
+                    break
+        return self._primary_obj.bbox
+
+    @roi.setter
+    def roi(self, value: BBox):
+        self._primary_obj.bbox.x_center = value.x_center
```

Comment:
_primary_obj can be None. 
```suggestion
        self.roi.x_center = value.x_center
```

Reply:
It can be None by definition, but our current pipeline logic relies on the existence of the primary object. So if we don't have the primary object we will get an error and it's OK.

Reply:
https://github.com/insight-platform/Savant/blob/41-roi-change/savant/deepstream/pyfunc.py#L27

If you try to use `frame_meta.roi = BBox(100,100,200,200)` in your pyfunc, you will encounter an error indicating that self._primary_obj is None. However, in the deepstream meta, it will be defined.

Reply:
fixed (the case is added to the test)